lib/saluki-core/src/topology/interconnect/event_buffer.rs

```diff
@@ -115,6 +141,22 @@ impl Iterator for IntoIter {
     }
 }
 
+#[allow(unused)]
+/// Predicate function for extracting metric events.
+pub fn is_metric(event: &Event) -> bool {
+    matches!(event.data_type(), DataType::Metric)
+}
+
+/// Predicate function for extracting eventd events.
+pub fn is_eventd(event: &Event) -> bool {
+    matches!(event.data_type(), DataType::EventD)
+}
+
+/// Predicate function for extracting service check events.
+pub fn is_service_check(event: &Event) -> bool {
+    matches!(event.data_type(), DataType::ServiceCheck)
+}
+
```

Comment:
I would just move these directly to `Event` and have them match on the structure itself (i.e. `matches!(self, Event::EventD(_))`).

Simple enough as a basic helper method and this module is kind of an obscure place for them to leave, even if we primarily use them here.